### PR TITLE
Upgrade DPL to v2.3.0

### PR DESCRIPTION
**Pull request by macbre**
http://www.mediawiki.org/wiki/Extension:DynamicPageList_(third-party)

Backporting changes from https://git.wikimedia.org/history/mediawiki%2Fextensions%2FDynamicPageList (in sync with [d142617](https://git.wikimedia.org/commitdiff/mediawiki%2Fextensions%2FDynamicPageList/d142617a0904fc8b70a8180e9f1653e85ba848b9))

@wladekb / @michalroszka / @owend

### CODE REVIEW

File: extensions/DynamicPageList/DPLMain.php [hunk 1]
```diff
@@ -2183,20 +2176,27 @@ public static function dynamicPageList( $input, $params, $parser, &$bReset, $cal
...
-            $sSqlCond_page_rev .= ' AND ' . $dbr->addQuotes($sNotModifiedBy) . ' NOT IN (SELECT rev_user_text FROM '.$sRevisionTable.' WHERE '.$sRevisionTable.'.rev_page=page_id)';
-        }
+			$sSqlCreationRevisionTable = $sRevisionTable . ' AS creation_rev, ';
+			$sSqlCond_page_rev .= ' AND ' . $dbr->addQuotes($sCreatedBy) . ' = creation_rev.rev_user_text'
+								.' AND creation_rev.rev_page = page_id'
+								.' AND creation_rev.rev_parent_id = 0';
+		}
+		if ( $sNotCreatedBy != "" ) {
+			$sSqlNoCreationRevisionTable = $sRevisionTable . ' AS no_creation_rev, ';
+			$sSqlCond_page_rev .= ' AND ' . $dbr->addQuotes($sCreatedBy) . ' != no_creation_rev.rev_user_text'
```
Comment (wladekb) [inline extensions/DynamicPageList/DPLMain.php]: $sNotCreatedBy

File: extensions/DynamicPageList/DPLMain.php [hunk 2]
```diff
@@ -2183,20 +2176,27 @@ public static function dynamicPageList( $input, $params, $parser, &$bReset, $cal
...
+								.' AND no_creation_rev.rev_page = page_id'
+								.' AND no_creation_rev.rev_parent_id = 0';
+		}
+		if ( $sModifiedBy != "" ) {
+			$sSqlChangeRevisionTable = $sRevisionTable . ' AS change_rev, ';
+			$sSqlCond_page_rev .= ' AND ' . $dbr->addQuotes($sCreatedBy) . ' = change_rev.rev_user_text'
```
Comment (wladekb) [inline extensions/DynamicPageList/DPLMain.php]: $sModifiedBy ?

File: extensions/DynamicPageList/DPLMain.php (hunk 1 above)
Comment (macbre) [inline extensions/DynamicPageList/DPLMain.php]: https://git.wikimedia.org/commitdiff/mediawiki%2Fextensions%2FDynamicPageList/8a4b08b5231c9c98056b56480241a1c8d337bb4f

**Comment by wladekb** [comment]
lgtm

**Comment by macbre** [comment]
```
Running 'WikiaMobileServiceTest'...
Fatal error: Class 'WikiaMobileService' not found in /var/lib/jenkins/workspace/unitTests_php_branches/extensions/wikia/WikiaMobile/tests/WikiaMobileServiceTest.php on line 17
```

Unit tests failing are not related to DPL changes

**Pull request merged**